File: app/src/main/java/com/focus/mob/FinSessionActivity.java
package com.focus.mob;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.FrameLayout;
import androidx.appcompat.app.AppCompatActivity;

import com.focus.mob.data.AppDatabase;
import com.focus.mob.data.SessionRecord;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

public class FinSessionActivity extends AppCompatActivity {

    private FrameLayout btnEmoji1, btnEmoji2, btnEmoji3;
    private String selectedMood = "Neutral";
    private long sessionDurationMs = 25 * 60 * 1000;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_fin_session);

        sessionDurationMs = getIntent().getLongExtra("DURATION_MS", 25 * 60 * 1000);

        btnEmoji1 = findViewById(R.id.btn_emoji_1);
        btnEmoji2 = findViewById(R.id.btn_emoji_2);
        btnEmoji3 = findViewById(R.id.btn_emoji_3);

        Button btnInsights = findViewById(R.id.btn_insights);
        Button btnHome = findViewById(R.id.btn_home);

        btnEmoji1.setOnClickListener(v -> selectEmoji(btnEmoji1, "Bad"));
        btnEmoji2.setOnClickListener(v -> selectEmoji(btnEmoji2, "Neutral"));
        btnEmoji3.setOnClickListener(v -> selectEmoji(btnEmoji3, "Great"));

        btnInsights.setOnClickListener(v -> {
            saveSessionToDb();
            Intent intent = new Intent(FinSessionActivity.this, StatsActivity.class);
            startActivity(intent);
        });

        btnHome.setOnClickListener(v -> {
            saveSessionToDb();
            Intent intent = new Intent(FinSessionActivity.this, HomeActivity.class);
            intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
            startActivity(intent);
        });
    }

    private void selectEmoji(FrameLayout btn, String mood) {
        resetEmojis();
        btn.setBackgroundResource(R.drawable.bg_emoji_selected);
        selectedMood = mood;
    }

    private void resetEmojis() {
        btnEmoji1.setBackgroundResource(R.drawable.bg_emoji_unselected);
        btnEmoji2.setBackgroundResource(R.drawable.bg_emoji_unselected);
        btnEmoji3.setBackgroundResource(R.drawable.bg_emoji_unselected);
    }

    private void saveSessionToDb() {
        int minutes = (int) (sessionDurationMs / (1000 * 60));
        
        SessionRecord record = new SessionRecord();
        record.durationMinutes = minutes;
        record.timestamp = System.currentTimeMillis();
        record.moodFeedback = selectedMood;
        record.goal = "Focus Session";
        record.ambiance = "Lumina Default";

        ExecutorService executor = Executors.newSingleThreadExecutor();
        executor.execute(() -> {
            AppDatabase db = AppDatabase.getDatabase(getApplicationContext());
            db.sessionDao().insert(record);
        });
    }
}

File: app/src/main/java/com/focus/mob/ui/session/FinSessionActivity.kt
package com.focus.mob.ui.session
import com.focus.mob.R

import com.focus.mob.ui.auth.*
import com.focus.mob.ui.main.*
import com.focus.mob.ui.onboarding.*
import com.focus.mob.ui.session.*
import com.focus.mob.utils.*

import android.content.Intent
import android.os.Bundle
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import com.focus.mob.data.AppDatabase
import com.focus.mob.data.repository.AuthRepository
import com.focus.mob.data.repository.SessionRepository
import com.focus.mob.databinding.ActivityFinSessionBinding
import com.focus.mob.ui.viewmodel.SessionViewModel
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class FinSessionActivity : AppCompatActivity() {

    private lateinit var binding: ActivityFinSessionBinding
    private var selectedMood = "Neutral"
    private var sessionDurationMs = 25 * 60 * 1000L

    private val sessionViewModel: SessionViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityFinSessionBinding.inflate(layoutInflater)
        setContentView(binding.root)

        sessionDurationMs = intent.getLongExtra("DURATION_MS", 25 * 60 * 1000L)

        binding.btnEmoji1.setOnClickListener { selectEmoji(it as android.widget.FrameLayout, "Bad") }
        binding.btnEmoji2.setOnClickListener { selectEmoji(it as android.widget.FrameLayout, "Neutral") }
        binding.btnEmoji3.setOnClickListener { selectEmoji(it as android.widget.FrameLayout, "Great") }

        binding.btnInsights.setOnClickListener {
            saveSessionToDb()
            startActivity(Intent(this, StatsActivity::class.java))
            fadeTransition()
        }

        binding.btnHome.setOnClickListener {
            saveSessionToDb()
            startActivity(Intent(this, HomeActivity::class.java).apply {
                flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
            })
            fadeTransition()
        }
    }

    private fun selectEmoji(btn: android.widget.FrameLayout, mood: String) {
        resetEmojis()
        btn.setBackgroundResource(R.drawable.bg_emoji_selected)
        selectedMood = mood
    }

    private fun resetEmojis() {
        binding.btnEmoji1.setBackgroundResource(R.drawable.bg_emoji_unselected)
        binding.btnEmoji2.setBackgroundResource(R.drawable.bg_emoji_unselected)
        binding.btnEmoji3.setBackgroundResource(R.drawable.bg_emoji_unselected)
    }

    private fun saveSessionToDb() {
        val minutes = (sessionDurationMs / (1000 * 60)).toInt()
        sessionViewModel.saveSession(minutes, selectedMood)
    }
}
